### ISSUE TITLE
Accessibility: Status message not announced by screen readers [Product Archive Results Count]

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52131]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The "Showing 1–16 of 21 results" message is visually displayed, but screen readers do not announce it because it has not been programmatically defined as a status message. This is related to #WooCore-060.

![Image](https://github.com/user-attachments/assets/fbab18e2-3993-4fcd-81b3-16d889df2412)

### Component or Page: 

Component: Product Archive Results Count Block     

### Name affected element:

[Status message "Showing 1–16 of 21 results"](https://drive.google.com/file/d/1Vqfk01GRXmFx5af9aDqq81Yuy5bdpI9o/view?usp=drive_link)

### Relevant code snippets:

```
<div data-block-name=""woocommerce/product-results-count"" class="" woocommerce wc-block-product-results-count wp-block-woocommerce-product-results-count"" data-wc-navigation-id=""wc-product-results-count-0"" style=""""><p class=""woocommerce-result-count"">
	Showing 1–16 of 21 results</p>
</div>
```

### Recommended fix:

The status should be announced by the screen reader. Make sure any status message displayed as visible text has the appropriate ARIA role:
 
 1. If the status message contains important, time-sensitive information that should be communicated to users immediately (potentially clearing the speech queue of previous updates), use role=""alert"", which has an implicit aria-live value of assertive.
 2. Otherwise, use a role with an implicit aria-live value of polite:
  a. Use role=""status"" for a simple status message that's not urgent enough to justify interrupting the current task.
  b. Use role=""log"" if new information is added to the status message in meaningful order, and old information might disappear (such as a chat log, game log, error log, or messaging history).

**Comment by amberhinds** [comment]
@vikrampm1 @Manussakis - I see the role alert on the results, but with VoiceOver on and all of the other things that happen on the page with these filters, it doesn't announce the change. Let me know if we should open a new issue for this.

When the price filter block is on the shop page, VoiceOver just reads out the new maximum price. That's what this video shows:

https://github.com/user-attachments/assets/b3b6fa10-3e0b-4f4d-abf1-8ca4e8a93ab9

When the price filter is not on the page, it doesn't read anything at all. Also, when the page reloads, focus is lost and reset at the top of the page. Here's a video that shows the behavior without the price filter.

https://github.com/user-attachments/assets/1271512d-2541-4713-90c7-380b354e50a1

**Comment by Manussakis** [comment]
@amberhinds Thanks for testing this change.

Your test steps don't seem aligned with the ones in the PR description. The filter blocks were not considered during the development of this task. In addition to that, the Woo team is hesitant to approve changes on these blocks as they [will be replaced](https://github.com/woocommerce/woocommerce/pull/53610#issuecomment-2553284625) by new blocks soon.

Plus, once [this other PR was merged](https://github.com/woocommerce/woocommerce/pull/55338#issuecomment-2790226283), only the announcement after the first load should be voiced by screen readers. The Woo team is aware of it per [this comment](https://github.com/woocommerce/woocommerce/pull/55338#pullrequestreview-2755897197).

cc: @vikrampm1

**Comment by amberhinds** [comment]
@Manussakis - This issue was opened when testing with the old filter blocks an https://woo.equalizedigital.com/shop/ (password: wooa11y)

I do see now that the linked PR, https://github.com/woocommerce/woocommerce/pull/55338, only mentions pagination and initial page load, not on filtering. Thanks, too for bringing that comment to my attention.

The comment you linked describes a WCAG failure, not a preferred behavior. If filters are used, screen reader users need an auditory announcement that tells them the filtering was successful and how results have changed on the page.

This issue should be reopened and addressed. If not for the current filter blocks, it should be tested for the new filter blocks coming in 9.9.

**Comment by Manussakis** [comment]
Thanks for your feedback @amberhinds !

For me, it makes more sense to keep this issue closed, since it was created when testing with the old filter blocks, and create a new one when the new filter blocks come out, in case they have the same issue.

Please let me know your thoughts. 🙂

cc: @vikrampm1
